src/use-swr.ts

```diff
@@ -215,6 +215,18 @@ function useSWR<Data = any, Error = any>(
   const unmountedRef = useRef(false)
   const keyRef = useRef(key)
 
+  // do unmount check for callbacks
+  const eventsRef = useRef({
```

Comment:
why not use `useMemo` in here, no reference scene

Reply:
@thoamsy useMemo is more like to use for recalculation when deps changed. 
for the events trigger, the refer here is just creating an object which is never changed during renders, without any deps required. so I feel ref is more proper for this case : )